File: src/main/java/co/gatedaccess/web/data/repo/DeviceRepo.java
package co.gatedaccess.web.data.repo;

import co.gatedaccess.web.data.model.Device;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface DeviceRepo extends MongoRepository<Device, String> {
    Device findDeviceByMemberId(String memberId);
}

File: src/main/kotlin/co/gatedaccess/web/data/repo/DeviceRepo.kt
package co.gatedaccess.web.data.repo

import co.gatedaccess.web.data.model.Device
import org.springframework.data.mongodb.repository.MongoRepository

interface DeviceRepo : MongoRepository<Device?, String?> {
    fun findDeviceByMemberId(memberId: String?): Device?
}
